projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid.pipes.ts

```diff
@@ -234,18 +234,27 @@ export class IgxTreeGridPagingPipe implements PipeTransform {
     }
 
     public transform(collection: ITreeGridRecord[], page = 0, perPage = 15, id: string, pipeTrigger: number): ITreeGridRecord[] {
-        if (!this.gridAPI.get(id).paging) {
+        const grid: IgxTreeGridComponent = this.gridAPI.get(id) as IgxTreeGridComponent;
+        if (!grid.paging) {
             return collection;
         }
 
+        const len = collection.length;
+        let totalPages = 1;
+
+        if (collection.length >= 0) {
+            totalPages = Math.ceil(len / perPage);
+        }
+
         const state = {
-            index: page,
+            index: (page >= totalPages - 1) ? totalPages - 1 : page,
```

Comment:
There is an issue when setting a filter that filters out all records. I would suggest to modify the code like that:
```
const totalPages = Math.ceil(len / perPage);
const state = {
    index: (totalPages > 0 && page >= totalPages) ? totalPages - 1 : page,
    recordsPerPage: perPage
};
```